mongo/integration/unified/client_entity.go

```diff
@@ -528,38 +549,40 @@ func setClientOptionsFromURIOptions(clientOpts *options.ClientOptions, uriOpts b
 	var wcSet bool
 
 	for key, value := range uriOpts {
-		switch key {
+		switch strings.ToLower(key) {
```

Comment:
Suggest using case insensitive matching. The motivation for this is that the "[failCommand](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#failcommand)" specifications indicate that the field in spec tests should be "appName". However, there doesn't seem to be consistency in the unified spec tests, often "appname" is used, especially in logging. This update will ensure more consistent behavior.